### ISSUE TITLE
Blazor Web View Authentication?

### ISSUE THREAD

**Issue by jhancock4d** [xamarin/MobileBlazorBindings#153]
Does the standard blazor web assembly library work for authentication when using Blazor Web View? If not, this needs to be adapted because this is basically required for any real-world dev now.

**Comment by Eilon** [comment]
@jhancock4d good question. I'm not certain, but I'd be surprised if it would "just work." It could be that some things would work, but perhaps not many. It presumably depends on what auth flow is used. And then what you'd actually want to happen in a native/hybrid app is probably different from what you want in a web app. For example, where are cookies stored? Or OAuth tokens?

Ultimately this is an area that will need some deep exploration. Auth is unfortunately never truly easy, and hybrid apps surely don't make it easy, either.

If you end up trying something out, I'd love to know what the experience was and what you would find useful that we could ship as a feature.

**Comment by jhancock4d** [comment]
So, it's absolutely vital that we have a cross platform "just works" OpenIdConnect compliant system built in.  This isn't just us, this IS the standard.  And developers get it wrong endlessly, so it's vital that it comes from Microsoft and is baked in and transparent like the Blazor auth for web assembly.

I'd note, in hopes that somehow this gets to the Edge team, that the browser should support exactly the same flow and security for sites as native apps. That is, javascript should allow kick off of OpenIdConnect that pops a locked window for authentication against the IdP passing in the config endpoint and client id, and then it handles the entire auth flow in that window as if it was native based on the url, client id and idp. And also handles refresh tokens as well and encrypts the entire thing and is opaque to javascript with the exception of providing the Access Token and Identity Token so that it's entirely secure and can't be accessed by javascript.  If this functionality was added to Chromium and Firefox also adopted it would solve the fundamental insecurities of OpenIdConnect and web and end the iframe hidden session refresh stuff that is a nightmare while also enabling safe use of refresh tokens too.

So a 1, 2 punch of Blazor getting this right and Edge implementing this right and giving it to Chromium would be a huge win for security on browsers, apps and the world (not exagerating) if done right.

**Comment by ktaze1** [comment]
I set up a [new project](https://github.com/ktaze1/BlazorHybridExperiment) with Blazor Hybrid and added a .NET Core Web Application to use it as a server. In server DbContext and ApplicationUser classes inherits Identity as usual. On Client side I added a custom AuthenticationStateProvider. Running multiple projects (Android and Server in my case) worked for me. So far I can use `<AuthorizeView>` without any problem. 

Although it worked, while trying to make the authentication work I had to allow the Cleartext Traffic on Android. I don't know how of this a security issue.
